apps/visualization/rest_filters.py

```diff
@@ -23,6 +25,35 @@ def filter_topic(self, qs, name, value):
         return qs.filter(topic__in=topics)
 
 
+class SourceListAggFilter(filters.FilterSet):
+    emergency = filters.CharFilter(field_name='emergency')
+    iso3 = filters.CharFilter(field_name='iso3', method='filter_iso3')
+    source_date_from = filters.DateFilter(field_name="source_date", lookup_expr='gte')
+    source_date_to = filters.DateFilter(field_name="source_date", lookup_expr='lte')
+
+    class Meta:
+        model = SourceListAgg
+        fields = ()
+
+    def filter_iso3(self, qs, name, value):
+        if not value:
+            return qs
+        return qs.filter(iso3__icontains=value)
+
+    @property
+    def qs(self):
+        qs = super().qs
+        query_params = [k for k, v in self.request.query_params.items() if not v == '']
+        # For date range, if not mentioned, default action is to pull the data from the last 7 days.
+        if not any(item in query_params for item in ['date_from', 'date_to']):
```

Comment:
date_from = request.GET.get('date_from')
date_to = request.GET.get('date_from')

if not (date_from or date_to):